geode-for-redis/src/main/java/org/apache/geode/redis/internal/netty/ExecutionHandlerContext.java

```diff
@@ -232,7 +236,14 @@ public void channelInactive(ChannelHandlerContext ctx) {
     }
   }
 
-  private void executeCommand(Command command) throws Exception {
+  public void resubmitCommand(Command command) {
+    ChannelHandlerContext ctx =
+        channel.pipeline().context(ByteToCommandDecoder.class.getSimpleName());
+    ctx.fireChannelRead(command);
```

Comment:
do you know for sure that fireChannelRead is async? That is, it just hands command off to some queue and does not end up processing command in this thread.

Reply:
99% sure :). I think the only time it might not be the case is when the current thread is also the same thread that is associated with the original blocking command's ExecutionHandlerContext.